### PR TITLE
Add custom config support

### PR DESCRIPTION
**Pull request by artem-zinnatullin**

### CODE REVIEW

File: detekt/detekt.bzl [hunk 1]
```diff
@@ -4,35 +4,43 @@ Detekt is a static analysis tool for Kotlin: https://github.com/arturbosch/detek
 """
 
 def _impl(ctx):
+    action_inputs = [] + ctx.files.srcs
+
+    # See Detekt CLI documentation: https://arturbosch.github.io/detekt/cli.html
     # TODO: Allow customizing JVM options.
-    arguments = [
+    action_arguments = [
```
Comment (kastiglione) [inline detekt/detekt.bzl]: Did you consider an [`Args`](https://docs.bazel.build/versions/master/skylark/lib/Args.html) object?

```python
action_arguments = ctx.actions.args()
action_arguments.add_all([
    "-Xms16m",
    "-Xmx128m",
    "-jar",
    ctx.file._detekt_cli_jar.path,
])
```

File: detekt/detekt.bzl [hunk 2]
```diff
@@ -43,12 +51,13 @@ detekt = rule(
         # Later we should allow user to customize Detekt binary.
         "_detekt_cli_jar": attr.label(default = "@detekt_cli_jar//file", allow_single_file = True),
         "srcs": attr.label_list(allow_files = True),
-        "config": attr.label(default = None, allow_single_file = True),
+        "config": attr.label(default = None, allow_single_file = True, doc = "Custom Detekt config file (must end with .yml). If not set Detekt will use its default configuration (mind the Detekt version) https://github.com/arturbosch/detekt/blob/master/detekt-cli/src/main/resources/default-detekt-config.yml"),
+        # TODO: Baselines are not fully supported yet due to Detekt relying on absolute paths which doesn't work with Bazel sandboxing.
+        "_baseline": attr.label(default = None, allow_single_file = True),
```
Comment (kastiglione) [inline detekt/detekt.bzl]: How does this private attr get set if it has no default?

Comment (artem-zinnatullin) [reply detekt/detekt.bzl]: I guess `None` is valid value? :D I actually want it to be `None` by default

Similar with `config` btw

File: detekt/detekt.bzl (hunk 1 above)
Comment (artem-zinnatullin) [reply detekt/detekt.bzl]: ah nice

File: detekt/detekt.bzl (hunk 2 above)
Comment (kastiglione) [reply detekt/detekt.bzl]: I thought private attrs could not by set through rule invocation, in which case the value is always None.

**Pull request merged**